### DISCUSSION TITLE
Form isDirty issue (antd)

### DISCUSSION THREAD

**Discussion by kaisiant** [react-hook-form/react-hook-form#3152] [Q&A] [answered]
Hello, I need some help about the form state. I am using form provider with antd UI.
I realize if one of the fields does not set default values and the form state isDirty will be invalid even if I just editing on the field that have default value.

CSB: https://codesandbox.io/s/react-hook-form-formstate-dirty-touched-submitted-forked-dk0lb?file=/src/RandomComponent.js

First input has default value
Second input no default value
```
1. type "a" into the first input => isDirty: true
2. delete "a" from the first input => isDirty: true
```

Expected result: 
`delete "a"  from the first input  => isDirty: false`

**Comment by bluebill1049** [comment, marked as answer]
```
defaultValues: {
      test: "",
      testNoDefaultValues: '',
    }```

**Reply by kaisiant** [reply]
Is it possible to not set the default values of all fields and `isDirty` is still working as expected.

**Reply by bluebill1049** [reply]
you need to provide a defaultValues or defaultValue at controller, otherwise, there is nothing for hook form to compare against.

**Comment by Joshuajrodrigues** [comment]
This doesn't work with async data when i need to do form.setFieldsValue

**Reply by BrendanC23** [reply]
Can you provide a CodeSandbox reproduction?
